lib/modules/manager/tekton/extract.ts

```diff
@@ -0,0 +1,98 @@
+import is from '@sindresorhus/is';
+import { loadAll } from 'js-yaml';
+import { logger } from '../../../logger';
+import { getDep } from '../dockerfile/extract';
+import type { PackageDependency, PackageFile } from '../types';
+import type { TektonBundle, TektonResource } from './types';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string
+): PackageFile | null {
+  logger.trace('tekton.extractPackageFile()');
+  const deps: PackageDependency[] = [];
+  let docs: TektonResource[];
+  try {
+    docs = loadAll(content) as TektonResource[];
+  } catch (err) {
+    logger.debug(
+      { err, fileName },
+      'Failed to parse YAML resource as a Tekton resource'
+    );
+    return null;
+  }
+  for (const doc of docs) {
+    deps.push(...getDeps(doc));
+  }
+  if (!deps.length) {
+    return null;
+  }
+  return { deps };
+}
+
+function getDeps(doc: TektonResource): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+  if (is.falsy(doc)) {
+    return deps;
+  }
+
+  // Handle TaskRun resource
+  addDep(doc.spec?.taskRef, deps);
+
+  // Handle PipelineRun resource
+  addDep(doc.spec?.pipelineRef, deps);
+
+  // Handle Pipeline resource
+  for (const task of doc.spec?.tasks ?? []) {
+    addDep(task.taskRef, deps);
+  }
+
+  // Handle TriggerTemplate resource
+  for (const resource of doc.spec?.resourcetemplates ?? []) {
+    addDep(resource?.spec?.taskRef, deps);
+    addDep(resource?.spec?.pipelineRef, deps);
+  }
+
+  // Handle list of TektonResources
+  for (const item of doc.items ?? []) {
+    deps.push(...getDeps(item));
+  }
+
+  return deps;
+}
+
+function addDep(ref: TektonBundle, deps: PackageDependency[]): void {
+  if (is.falsy(ref)) {
+    return;
+  }
+  let imageRef = '';
+  // Find a bundle reference from the Bundle resolver
+  if (ref.resolver === 'bundles') {
+    for (const field of ref.resource ?? []) {
+      if (field.name === 'bundle') {
+        imageRef = field.value;
+        break;
+      }
+    }
+  }
+
+  if (is.emptyStringOrWhitespace(imageRef)) {
```

Comment:
```suggestion
  if (!imageRef || is.emptyStringOrWhitespace(imageRef)) {
```
so we need the additional check again

Reply:
This will not catch the case where imageRef is not a string, but a truthy value. If we want a proper check, we should revert back to using `is.string` since TypeScript does not perform type checking at runtime. We can also create a comprehensive type guard function. This seems sort of heavy handed, especially because I have not seen this approach in other places of the code base.

I quite like the current approach as it highlights that an attribute was not processed because it has an unexpected type, which I believe is what `invalid-value` is meant to cover.

Reply:
ok, then we should somplify this here to:
```suggestion
  if (is.nullOrUndefined(imageRef)) {
```

is `imageRef` is isn't defined, fallback to old style.

Reply:
That doesn't handle when the string is empty, which results in a dependency like this being detected:
```typescript
{
  "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
  "currentDigest": undefined,
  "currentValue": undefined,
  "datasource": "docker",
  "depName": "",
  "depType": "tekton-bundle",
  "replaceString": "",
},
```

I'll be honest that I don't quite understand why we want to simplify this further. As mentioned in the previous comment, using `is.emptyStringOrWhitespace` seems to provide the most correct behavior, and it's already a single call.

Reply:
@viceice, any additional feedback here? :pray:

Reply:
it should not fallback if `imageRef` is defined, even it's not a string value.

although the current returned invalid values aren't helpful, as you don't know where they com from. so we should filter any non string and empty or whitespace strings.

Reply:
ok, I think I understand your concern now. thank you for clarifying. Updated the code.